src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,224 @@
+                    }
+
+                // Current task is an Event
+                } else if (taskString[0].equals("E")) {
+                    String[] periodDetails = taskString[2].split(" ");
+                    if (periodDetails.length == 3) {
+                        newTask = new Event(periodDetails[0],
+                                LocalDate.parse(periodDetails[1]), LocalDate.parse(periodDetails[2]));
+                    } else if (periodDetails.length == 4) {
+                        newTask = new Event(periodDetails[0], LocalDate.parse(periodDetails[1]),
+                                LocalTime.parse(periodDetails[2]), LocalTime.parse(periodDetails[3]));
+                    } else if (periodDetails.length == 5) {
+                        newTask = new Event(periodDetails[0],
+                                LocalDate.parse(periodDetails[1]), LocalTime.parse(periodDetails[2]),
+                                LocalDate.parse(periodDetails[3]), LocalTime.parse(periodDetails[4]));
+                    } else {
+                        throw new DukeException(DukeExceptionType.DB_READ);
+                    }
+
+                // Current task is a Todo
+                } else if (taskString[0].equals("T")) {
+                    newTask = new Todo(taskString[2]);
+
+                // Current task is in invalid format
+                } else {
+                    throw new DukeException(DukeExceptionType.DB_READ);
+                }
+
+                // Handles whether or not Task is done
+                if (Integer.parseInt(taskString[1]) == 1) {
+                    newTask.setDone();
+                }
+
+                // Add Task to task list, continue with next Task
+                savedTasks.add(newTask);
+                currLine = reader.readLine();
+            }
+
+        } catch (DukeException e) {
+            ui.showException(e);
+
+        } catch (IOException e) {
+            ui.showException(new DukeException(DukeExceptionType.DB_LAUNCH));
+        }
+
+        return savedTasks;
+    }
+
+    /**
+     * Adds the given entry to the database (hard disk).
+     *
+     * @param s A string representation of the task to be added to the database.
+     */
+    public void addDbEntry(String s) {
+        try {
+            FileWriter writer = new FileWriter(duke, true);
+            writer.write(s + "\n");
+            writer.close();
+
+        } catch (IOException e) {
+            ui.showException(new DukeException(DukeExceptionType.DB_ADD));
+        }
+    }
+
+    /**
+     * Sets the given entry in the database to 'done'.
+     *
+     * @param s A string representation of the task in the database to be set to 'done'.
+     */
+    public void setDbEntryDone(String s) {
+        try {
+            // Create updated file
+            String updatedPath = filePath.split("/")[0] + "/updated.txt";
+            File updated = new File(updatedPath);
```

Comment:
Maybe use a more descriptive name like updatedFile as it might get confusing in the later lines as to what "updated" is.